### PR TITLE
Create new org and admin

### PR DESCRIPTION
**Pull request by kasugaijin**
# 🔗 Issue

[175](https://github.com/rubyforgood/pet-rescue/issues/175)

# ✍️ Description

This PR adds a service object that we can use via the console to:
- create a new organization
- create a new user
- create a new staff account
- add the admin role
- send email to admin user

There is also a new mailer and template for this flow.

# 📷 Screenshots/Demos

### CODE REVIEW

File: app/mailers/sign_up_mailer.rb
Comment (kasugaijin) [inline app/mailers/sign_up_mailer.rb:32]: Not sure what to use as the `from` email. I suspect we'd want this to be from pet rescue rather than the org itself.

File: app/services/create_org_and_admin_service.rb
Comment (Ptrboro) [inline app/services/create_org_and_admin_service.rb:84]: Looks like the password is not needed here. The mailer uses `@user.password` instead.

```suggestion
    OrganizationMailer.with(
      user: @user,
      organization: @organization
    )
```

**Review by Ptrboro — APPROVED** [review]
LGTM! Leaving one suggestion

File: app/services/create_org_and_admin_service.rb
Comment (edwinthinks) [inline app/services/create_org_and_admin_service.rb:2]: Hey @kasugaijin, I've got a few ideas that I want to first say are suggestions and ideas for how to improve this. I'am a big believer in coming back to improve rather than put PRs in a holding pattern. With that said, here are some considerations and ideas to counter-act them.

Consider the situation that maybe 1 of the 4 actions failed. You'll need to re-run that 1 command until it works if you realize it didn't work. This can be problematic because the data for the organization will be in a "invalid" state with missing associations or records. One way to prevent this is issue is wrapping the entire process inside of a transaction block (https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html). The transaction block lets you insert and update in seperate lines but rollbacks if there is an issue. I would use this transaction and use a signal method that takes a hash of information to be able to run.

Let me know if that makes sense

File: app/services/create_org_and_admin_service.rb
Comment (edwinthinks) [inline app/services/create_org_and_admin_service.rb:1]: We may find ourselves creating more services like this. I think renaming this to be Organizations::CreateService would start allowing us to namepace these services to keep things organized.

File: app/services/create_org_and_admin_service.rb
Comment (edwinthinks) [inline app/services/create_org_and_admin_service.rb:97]: Related to my comment above about wrapping everything in a single transaction. We could then remove this likely.

**Review by edwinthinks — CHANGES_REQUESTED** [review]
Thanks for working on this @kasugaijin let me know if you have any Qs!

File: app/services/organizations/create_service.rb
Comment (edwinthinks) [inline app/services/organizations/create_service.rb:22]:
```suggestion
    raise "An error occurred: #{e.message}"
```
So it is really loud when the error occurs.

**Review by edwinthinks — APPROVED** [review]
I think after the small suggestion we can get this merged in.

P.S - I'am curious about using https://github.com/collectiveidea/interactor to create a common service interface. I don't want to hold off the PR any longer so I think we can update it later or if you want to explore the use of the the gem.

**Review by edwinthinks — APPROVED** [review]
LGTM!

**Pull request merged**
